src/components/inputs/SingleSelect.tsx

```diff
@@ -0,0 +1,127 @@
+import { ReactNode, useEffect, useState, useRef } from "react";
+import PopoverButton from "../buttons/PopoverButton/PopoverButton";
+import { Item } from "./checkboxes/CheckboxList";
+import { css } from '@emotion/react';
+import { uniqueId } from "lodash";
+  
+export interface SingleSelectProps {
+    items: Item[];
+    value: string;
+    onSelect: (value: string) => void;
+    buttonDisplayContent: ReactNode;
+}
+
+export default function SingleSelect({
+    items,
+    value,
+    onSelect,
+    buttonDisplayContent,
+}: SingleSelectProps) {
+    const [ isPopoverOpen, setIsPopoverOpen ] = useState<boolean>(false);
+    const [ focusedElement, setFocusedElement ] = useState(
+        items.findIndex(item => value === item.value) !== -1 ? items.findIndex(item => value === item.value) : 0
```

Comment:
I think this expression could be moved to a variable assignment, and then reused in the `useEffect` below. It would also be a good idea to only call `items.findIndex(...)` once. This would turn what is currently 4 iterations of `items` to 1.